src/Test.UnitTests.Sarif/RoundTrippingTests.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information. 
+
+using System;
+using System.Globalization;
+using FluentAssertions;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Sarif
+{
+    public class RoundTrippingTests
+    {
+        private static ResourceExtractor s_extractor = new ResourceExtractor(typeof(RoundTrippingTests));
+
+        private static string GetResourceContents(string resourceName)
+            => s_extractor.GetResourceText($"RoundTripping.{resourceName}");
+
+        // Class used to test serialization and deserialization of an arbitrary class.
+        public class CustomObject
+        {
+            public DateTime DateTime { get; set; }
+            public string String { get; set; }
+        }
+
+        private static DateTime ParseUtcDateTime(string dateTime)
+        {
+            // DateTimeStyles.AdjustToUniversal is necessary because otherwise DateTime.Parse
+            // adjusts the time from UTC to local time (for example, in Redmond, it subtracts
+            // 7 hours). We want the hours field to be exactly as specified in the test file.
+            return DateTime.Parse(dateTime, provider: null, DateTimeStyles.AdjustToUniversal);
+        }
+
+        [Fact]
+        public void SarifLog_PropertyBagProperties_CanBeRoundTripped()
+        {
+            string originalContents = GetResourceContents("RoundTripping.sarif");
+
+            SarifLog log = SarifUtilities.DeserializeObject<SarifLog>(originalContents);
+
+            PropertyBagHolder holder = log.Runs[0].Results[0];
+
+            DateTime dateTimeProperty = holder.GetProperty<DateTime>("dateTime");
+            dateTimeProperty.Should().Be(ParseUtcDateTime("2019-09-25T15:07:01Z"));
+
+            string stringProperty = holder.GetProperty("string");
+            stringProperty.Should().Be("Hello, string!");
+
+            int intProperty = holder.GetProperty<int>("int");
+            intProperty.Should().Be(42);
+
+            long longProperty = holder.GetProperty<long>("long");
+            longProperty.Should().Be(5000000000);
```

Comment:
Comment affirming that this example is over int.MaxValue?

Reply:
That comment appears on Line 55.

---
In reply to: [329664092](https://github.com/microsoft/sarif-sdk/pull/1688#discussion_r329664092) [](ancestors = 329664092)